### ISSUE TITLE
[BLOCKED] try to make CompleteMoabHandler less confusing

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/preservation_catalog#366]
Per discussion on #1487, start small.  E.g., are there knotty nested conditionals in long methods, where some of the nested logic could be extracted into a nicely named helper method?  If we chip away at that sort of thing, hopefully the class becomes more approachable, and easier to test and refactor.

A larger sort of refactor might be splitting the class into multiple classes for different categories of work (e.g. validating a moab we know is in the catalog, vs adding a moab we haven't seen before to the catalog vs updating a moab we know is in the catalog when it gets versioned, etc).

It might be easiest to spawn smaller more concrete tickets from this, leaving this open as a catch all, as this will probably require a few iterations to get things into a satisfactory state.

~~It would probably be best to do the `MoabValidationHandler` refactoring first (whichever we keep of #1276 and #1494, since those are dupes).  Refactoring `CompleteMoabHandler` and `MoabValidationHandler` concurrently sounds like a recipe for rebase hell.~~

As a start, maybe look at the class and note where things get hard to follow, or where intent/purpose is hard to discern.  E.g. when your eyes glaze over on a super long method or a 3 levels of nested conditionals, or whatever.

**Comment by jmartin-sul** [comment]
https://github.com/sul-dlss/preservation_catalog/pull/1294 might be able to provide some inspiration

**Comment by ndushay** [comment]
@jmartin-sul I'm closing this ticket as it is 5 years old and I think the refactors done recently have helped a lot.
